### DISCUSSION TITLE
Keep attrributes of thematic layer in output layer

### DISCUSSION THREAD

**Discussion by maarten-vermeyen** [OnroerendErfgoed/brdr#34] [Ideas]
Idea from @MichielVansteenkiste:

> Hi everyone,
> 
> I'm Michiel Vansteenkiste from the Flemish Environment Agency. We aim to use the BRDR script to accurately define the boundaries of zoning plans. I propose **retaining the original attributes** after running the script. Although some data like 'area' and 'length' might no longer be accurate, keeping the original attributes can be useful for reference and validation.
> 
> I look forward to your feedback!

**Comment by maarten-vermeyen** [comment]
Hi Everyone,

My first thoughts, just to kick of the discussion. 

The feature seems straightforward to implement, at least at first glance. It is simply a join with the attributes of the thematic layer. The ID of the thematic layer is already present in the output layers. However, there are some questions related to this feature and the implementation that should be considered.

1. The current script is implemented as a geoprocessing script. This means the functionality should be to limited to 'do one thing, and do it well'. In that regard, the join can be accomplished by daisy chaining our script and the `join attributes by field value` algorithm. Implementing the feature would create a functional overlap, that might infringe upon the stated design principle. However, this is (a lot) more complicated for end users for a relatively common use case. The high number of input variables makes it harder to create a custom model.
2. If we implement the feature, should it be done on all layers (including the diff layers) , or only the result layers?
3. If we implement the feature, should it be optional (my personal preference) or should it default to preserving attributes? Should we have a custom prefix for output layers to avoid name collisions?

In the mean time, the easiest way to rejoin attributes in a GUI user scenario, is to create a new join in the 'Joins' tab of the result layer properties. This is a relatively straightforward operation, and we could add a section to the documentation. This could be sufficient for occasional users, but the extra clicks might get tiresome for people who use this a lot.

Kind regards,

Maarten

**Comment by dieuska** [comment]
Hi, in the last released version of brdrQ (v 0.9.8), a parameter (checkbox) is added  to keep the original layer-attributes in the result: https://github.com/OnroerendErfgoed/brdrQ/blob/main/autocorrectborders.py

**Comment by koenedaele** [comment]
I assume this is now working as requested by @MichielVansteenkiste. Closing this.
